src/main/python/tensorframes/core.py

```diff
@@ -377,6 +378,23 @@ def analyze(dframe):
     """
     return DataFrame(_java_api().analyze(dframe._jdf), _sql)
 
+def append_shape(dframe, col, size):
+    """Append extra metadata for a dataframe that
+     describes the numerical shape of the content.
+
+     This method is useful when a dataframe contains non-scalar tensors, for which the shape must be checked beforehand.
```

Comment:
Can you add a comment also that indicates that the user is responsible for providing the right shape, and that a mismatch will trigger eventually an exception in Spark?